src/NServiceBus.Core/Unicast/Transport/TransportConnectionString.cs

```diff
@@ -2,35 +2,40 @@
 {
     using System;
     using System.Configuration;
-    using Transport;
 
-    sealed class TransportConnectionString
+    /// <summary>
+    /// Allows to get the configured transport connection string.
+    /// </summary>
+    public sealed class TransportConnectionString
     {
         TransportConnectionString()
         {
         }
 
+        /// <summary>
+        /// Creates new connection string configuration that returns value provided by a callback.
+        /// </summary>
+        /// <param name="func">Callback providing connection string value.</param>
         public TransportConnectionString(Func<string> func)
```

Comment:
OT: I'd think we should let the transports provide the connection string api individually in vNext, all this cruft in the core isn't worth the gain. https://github.com/Particular/NServiceBus/issues/4297